riscv/src/compiler.rs

```diff
@@ -904,6 +904,27 @@ fn process_instruction(instr: &str, args: &[Argument]) -> Vec<String> {
             let (rd, r1, r2) = rrr(args);
             only_if_no_write_to_zero(format!("tmp1, {rd} <== mul({r1}, {r2});"), rd)
         }
+        "mulhsu" => {
+            let (rd, r1, r2) = rrr(args);
+            only_if_no_write_to_zero_vec(
+                vec![
+                    format!("tmp1 <== to_signed({r1});"),
+                    // tmp2 is 1 if tmp1 is non-negative
+                    "tmp2 <== is_positive(tmp1 + 1);".into(),
+                    // If negative, convert to positive
+                    "skip_if_zero 0, tmp2;".into(),
```

Comment:
I know this was not introduced here, but I'm wondering if the skip instruction is actually compatible with the batching.

Reply:
But with smarter batching, if the second argument is always a constant, I think this can be handled.

Alternatively, instead of this relative jump instruction, we can add new labels an jump there.

Reply:
Sure, it is compatible with the current batching, we just need to be careful that future batching does not destroy it (and it shouldn't since people can do what they want with `pc`).